packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -326,6 +331,36 @@ function parseBaseElement(
     return element;
 }
 
+function parseLwcElement(
+    ctx: ParserCtx,
+    parse5Elm: parse5.Element,
+    parsedAttr: ParsedAttribute,
+    parse5ElmLocation: parse5.ElementLocation
+) {
+    const { tagName: tag, namespaceURI } = parse5Elm;
+
+    if (tag === LwcTagName.Component) {
```

Comment:
May be also error if the `<lwc:component>` tag contains `lwc:dynamic` directive?

Reply:
In the `applyLwcDynamicDirective` there's a [check](https://github.com/salesforce/lwc/pull/3337/files#diff-7d2d3ce772f6d0a96d43e3579efb79f9c91407e26851f4500db2c8181f5f4aa4R936-R940) that the directive can only be used on custom elements.

There's a similar [check](https://github.com/salesforce/lwc/pull/3337/files#diff-7d2d3ce772f6d0a96d43e3579efb79f9c91407e26851f4500db2c8181f5f4aa4R969-R971) in `applyLwcIsDirective` that it can only be used on a `<lwc:component>` element.